docker-compose.yml

```diff
@@ -1,4 +1,9 @@
 x-env-mapping: &env
+  # https://docs.docker.com/compose/environment-variables/envvars-precedence/
+  env_file:
```

Comment:
This is an important, but I think good change. This means if there is no .env file, docker compose will just fail. It would prevent someone from running anything via docker compose before running `make create_env_file`.

There is not really a scenario where you should run docker compose without it, it is critical for configuring the compose project.

happy to hear push back if you think this is too strong enforcement.

Reply:
Boo, docker compose version in gha is not compatible with this syntax. Not sure it's worth the effort to upgrade.